simtools/applications/generate_corsika_histograms.py

```diff
@@ -0,0 +1,425 @@
+
+     .. code-block:: console
+
+        simtools-generate-corsika-histograms --IACT_file /workdir/external/simtools/tests/\
+            resources/tel_output_10GeV-2-gamma-20deg-CTAO-South.corsikaio --png --ecsv
+            --event_2D_histograms zenith azimuth --event_1D_histograms total_energy
+
+
+    Expected final print-out message:
+
+    .. code-block:: console
+        INFO::generate_corsika_histograms(l358)::main::Finalizing the application.
+        Total time needed: 8s.
+"""
+
+import logging
+import time
+from pathlib import Path
+
+import numpy as np
+
+import simtools.util.general as gen
+from simtools import io_handler
+from simtools.configuration import configurator
+from simtools.corsika import corsika_output_visualize
+from simtools.corsika.corsika_output import CorsikaOutput
+
+logger = logging.getLogger()
+
+
+def _parse(label, description, usage):
+    """
+    Parse command line configuration
+
+    Parameters
+    ----------
+    label: str
+        Label describing the application.
+    description: str
+        Description of the application.
+    usage: str
+        Example on how to use the application.
+
+    Returns
+    -------
+    CommandLineParser
+        Command line parser object
+
+    """
+    config = configurator.Configurator(label=label, description=description, usage=usage)
+
+    config.parser.add_argument(
+        "--IACT_file",
+        help="Name of the CORSIKA IACT file from which to generate the histograms.",
+        type=str,
+        required=True,
+    )
+
+    config.parser.add_argument(
+        "--telescope_indices",
+        help="Name of the CORSIKA IACT file from which to generate the histograms.",
+        type=str,
+        required=False,
+        nargs='+',
+        default=None
+    )
+
+    config.parser.add_argument(
+        "--individual_telescopes",
+        help="if False, the histograms are filled for all given telescopes together, otherwise"
+             "one histogram is set for each telescope separately.",
+        action='store_true',
+        required=False,
+        default=False,
+    )
+
+    config.parser.add_argument(
+        "--hist_config",
+        help="ecsv file with the configuration parameters to create the histograms.",
+        type=str,
+        required=False,
+        default=None,
+    )
+
+    config.parser.add_argument(
+        "--output_directory",
+        help="Output directory where to save the histograms.",
+        type=str,
+        required=False,
+        default=None,
+    )
+
+    config.parser.add_argument(
+        "--png", help="Save histograms into png files.", action="store_true", required=False
+    )
+
+    config.parser.add_argument(
+        "--ecsv", help="Save histograms into ecsv files.", action="store_true", required=False
+    )
+
+    config.parser.add_argument(
+        "--event_1D_histograms",
+        help="Arguments from the CORSIKA event header to extract 1D histograms.",
```

Comment:
This has been used in every application, this is nothing new. If you type the following command you get the help text for every argument:
`python simtools/applications/generate_corsika_histograms.py --IACT_file tests/resources/tel_output_10GeV-2-gamma-20deg-CTAO-South.corsikaio --help`

Reply:
I changed the explanation but it was not possible to use 'choices' because at this point the instance was not yet defined (it is actually needed to access the available keys)

Reply:
This is not important so we can skip this, but just for my curiosity, which would be the keys that you would get from the instance? Because technically there is a way to get them without creating the instance.

Reply:
you can get the list from `self.all_event_keys`. I don't think we need to point to this attribute in the help message because it would not be possible to show the available keys anyways from the application.

But if there is a way to get that before creating a class instance then lets do it

Reply:
No, this requires reading the CORSIKA file so it's impossible. You can forget about it then.